File: js/tlg-lib/src/state/StateLocked.ts
import { State } from './State'
import { InternalContext } from '../context/InternalContext'
import { StorageContext } from '../context/StorageContext'
import { StateID } from './StateID'

export class StateLocked extends State {
    constructor (c:InternalContext, s:StorageContext) {
        super(c, s)
        this.context.setStatusText('Game over')
    }

    public getID (): number {
        return StateID.ID_LOCKED
    }
}

File: tlg/src/main/kotlin/state/StateLocked.kt
package state

import context.InternalContext
import context.PlatformContext

class StateLocked(c: InternalContext) : State(c) {
    override fun init(c: PlatformContext): State {
        return this
    }

    override val id = ID

    override fun toString(): String {
        return "Game over"
    }

    companion object {
        private const val ID = 4
    }
}
